src/main/java/br/edu/ufcg/computacao/eureca/backend/core/dao/scsvfiles/ScsvFilesDataAccessFacade.java

```diff
@@ -361,32 +361,36 @@ public StudentCurriculumProgress getStudentCurriculumProgress(String studentRegi
     public StudentPreEnrollment getStudentPreEnrollment(String courseCode, String curriculumCode, String studentRegistration) throws InvalidParameterException {
         Curriculum curriculum = this.getCurriculum(courseCode, curriculumCode);
         StudentCurriculumProgress progress = this.getStudentCurriculumProgress(studentRegistration);
-        // ToDo: mudar essa lógica para considerar o número ideal de créditos por tipo
-        int idealNumberOfCredits = getIdealNumberOfCredits(curriculum, progress);
+        int idealMandatoryCredits = this.getIdealMandatoryCredits(curriculum, progress);
```

Comment:
Acho que actualTerm e nextTerm estão usados de forma confusa. O código devia ser o seguinte:

int actualTerm = getActualTerm(curriculum, progress); // getActualTerm é o novo nome de getNextTerm(...)
int nextTerm = actualTerm + (progress.getEnrolledCredits() > 0 ? 1 : 0);
int idealMandatoryCredits = this.getIdealMandatoryCredits(nextTerm);
...

A função getActualTerm(curriculum, progress) (antiga getNextTerm(...) deve ser modificada para retornar i+1 dentro do if e expectedMinAccumulatedCredits.size() (ao invés de 8) fora do loop.

Reply:
Testei retornando i + 1, mas o resultado estava errado por +1. Por hora deixei o retorno como apenas 'i', mas acho bom investigar isso posteriormente, pode ser um erro de semântica em algum lugar no código